project/src/com/google/daggerqueryui/scripts/binding_graph.js

```diff
@@ -0,0 +1,212 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const bindingGraph = (function() {
+  /**
+   * A dictionary that maps a node's name to its identifier.
+   * @type {Map<string, number>}
+   */
+  const nodesToNumbers = new Map();
+
+  /**
+   * A data structure <a href="https://ww3.arb.ca.gov/ei/tools/lib/vis/docs/dataset.html">DataSet</a>
+   * that contains all the nodes from the shown subgraph.
+   */
+  const nodes = new vis.DataSet();
+
+  /**
+   * A data structure <a href="https://ww3.arb.ca.gov/ei/tools/lib/vis/docs/dataset.html">DataSet</a>
+   * that contains all the edges from the shown subgraph.
+   */
+  const edges = new vis.DataSet();
+
+  /**
+   * Constructs a simple name from the full node name.
+   *
+   * @example from com.google.List[com.google.Beach] it will construct List[Beach].
+   * @param {string} node a full name of the node
+   */
+  function getSimpleName(node) {
+    const firstIndex = node.indexOf('[');
+
+    if (firstIndex === -1) {
+      const components = node.split('.');
+      return components[components.length - 1];
+    }
+
+    const container = node.substring(0, firstIndex).split('.').slice(-1);
+    const content = getSimpleName(node.substring(firstIndex + 1, node.lastIndexOf(']')));
```

Comment:
What if the node name has multiple generic args, e.g. `com.google.Map[java.lang.String, com.google.Foo]`?

Alternative idea - how about doing a regex replace, so you don't have to worry about characters like `[`, `]`, `,`

```
const fullNameRegex = /(?:\w+\.)*(\w+)/g;
const simpleName = node.replaceAll(fullNameRegex, '$1');
return simpleName;
```

Reply:
😻 Thanks you, that's an awesome solution with the regex!